lib/src/license_detection/confidence.dart

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of 'license_detector.dart';
+
+/// An instance indicating a mismatch between unknown text and a known license.
+@sealed
+@visibleForTesting
+class LicenseMismatchException implements Exception {
+  final String message;
+
+  LicenseMismatchException(this.message);
+
+  @override
+  String toString() {
+    return message;
+  }
+}
+
+/// Computes the confidence of [knownLicense] matching with [unknownLicense] and
+/// returns an instance of [LicenseMatch].
+///
+/// Throws if an unaccepatable change was made or the calculated confidence is
```

Comment:
```suggestion
/// Throws [LicenseMismatchException] if an unaccepatable change was made or the calculated confidence is
```

For `Exception` subclasses we specify what is thrown and why, for `Error` subclasses we just say `throws if...`.

Because `Exception`s are special cases; and `Error`s are always programmer mistakes in the program.